plugin/src/main/kotlin/com/nishtahir/CargoBuildTask.kt

```diff
@@ -162,3 +176,86 @@ open class CargoBuildTask : DefaultTask() {
         }.assertNormalExitValue()
     }
 }
+
+// We don't want to invoke `rustc -Vv` a ton, but we can't invoke it at all without
+// the gradle project (despite this not being dependent on the project). As a result,
+// we need to do some locking in order to cache the result of that.
+//
+// This object exists to encapsulate that, it just exposes a `fun getValue(p: Project): String?`.
+//
+// TODO: Is there somewhere more natural to put configuration state like this?
+private object DefaultTargetTriple {
+    // Initialized once, write-protected by `writeLock`, but no lock needs to be held when reading.
+    @Volatile
+    private var knownValue: String? = null
+
+    // Protects `knownValue` during writing. No protection is needed during reading. We also hold it
+    // while computing the value, to avoid wasting work.
+    private val writeLock = Object()
+
+    /**
+     * If we already know the triple, return it. Otherwise, call out to rustc to get it.
+     *
+     * On macOS, this should output `x86_64-apple-darwin`, for example. Returns null on any problems.
+     *
+     * This requires the `project` since I'm not sure how else to execute commands, but
+     * isn't actually dependent on the project.
+     */
+    internal fun getValue(project: Project): String? {
```

Comment:
Strongly consider allowing to override this in the `local.properties` file (just like you can set targets, etc, there).  Then we don't require a fire-drill if this is broken; local builders can always set it by hand for a while.

Reply:
I did but I can't imagine it ever doing what someone wants. That said, I agree about not wanting a fire drill.